nutanix/data_source_nutanix_foundation_central_imaged_nodes_list.go

```diff
@@ -0,0 +1,313 @@
+	conn := meta.(*Client).FoundationCentral
+	req := fc.ImagedNodesListInput{}
+
+	if len, lenok := d.GetOk("length"); lenok {
+		req.Length = utils.IntPtr(len.(int))
+	}
+	if offset, offok := d.GetOk("offset"); offok {
+		req.Offset = utils.IntPtr(offset.(int))
+	}
+
+	if filter, fok := d.GetOk("filters"); fok {
+		filt := &fc.ImagedNodeListFilter{}
+		filter := filter.([]interface{})[0].(map[string]interface{})
+		filt.NodeState = utils.StringPtr(filter["node_state"].(string))
+		req.Filters = filt
+	}
+
+	resp, err := conn.ListImagedNodes(ctx, &req)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	imgNodes := flattenImagedNodes(resp.ImagedNodes)
+
+	d.Set("imaged_nodes", imgNodes)
+
+	if resp.Metadata != nil {
+		metalist := make([]map[string]interface{}, 0)
+		meta := make(map[string]interface{})
+		meta["length"] = (resp.Metadata.Length)
+		meta["offset"] = (resp.Metadata.Offset)
+		meta["total_matches"] = (resp.Metadata.TotalMatches)
+
+		metalist = append(metalist, meta)
+		d.Set("metadata", metalist)
+	}
+
+	d.SetId(resource.UniqueId())
+	return nil
+}
+
+func flattenImagedNodes(imgcls []*fc.ImagedNodeDetails) []map[string]interface{} {
+	imgClsList := make([]map[string]interface{}, len(imgcls))
+	if len(imgcls) > 0 {
+		for i, v := range imgcls {
+			imgClsList[i] = map[string]interface{}{
+				"aos_version":         v.AosVersion,
+				"api_key_uuid":        v.APIKeyUUID,
+				"available":           v.Available,
+				"block_serial":        v.BlockSerial,
+				"created_timestamp":   v.CreatedTimestamp,
+				"current_time":        v.CurrentTime,
+				"cvm_gateway":         v.CvmGateway,
+				"cvm_ip":              v.CvmIP,
+				"cvm_ipv6":            v.CvmIpv6,
+				"cvm_netmask":         v.CvmNetmask,
+				"cvm_up":              v.CvmUp,
+				"cvm_uuid":            v.CvmUUID,
+				"cvm_vlan_id":         v.CvmVlanID,
+				"foundation_version":  v.FoundationVersion,
+				"hypervisor_gateway":  v.HypervisorGateway,
+				"hypervisor_hostname": v.HypervisorHostname,
+				"hypervisor_ip":       v.HypervisorIP,
+				"hypervisor_netmask":  v.HypervisorNetmask,
+				"hypervisor_type":     v.HypervisorType,
+				"hypervisor_version":  v.HypervisorVersion,
+				"imaged_cluster_uuid": v.ImagedClusterUUID,
+				"imaged_node_uuid":    v.ImagedNodeUUID,
+				"ipmi_gateway":        v.IpmiGateway,
+				"ipmi_ip":             v.IpmiIP,
+				"ipmi_netmask":        v.IpmiNetmask,
+				"ipv6_interface":      v.Ipv6Interface,
+				"model":               v.Model,
+				"node_position":       v.NodePosition,
+				"node_serial":         v.NodeSerial,
+				"node_state":          v.NodeState,
+				"node_type":           v.NodeType,
+				"object_version":      v.ObjectVersion,
+				"supported_features":  flattenFeature(v.SupportedFeatures),
+				"latest_hb_ts_list":   flattenFeature(v.LatestHbTSList),
+				"hardware_attributes": flattenHardwareAttributes(v.HardwareAttributes),
+			}
+		}
+	}
+	return imgClsList
+}
+
+func flattenFeature(pr []*string) []string {
```

Comment:
Please check if we have exiting function for this in utils ?

Reply:
Tried existing utils functions but they were not outputting the correct result in format.